lib/cn/csched_sp3_work.c

```diff
@@ -521,6 +525,9 @@ sp3_work_wtype_length(
             vwlen += stats->kst_vwlen;
             wlen += stats->kst_kwlen + stats->kst_vwlen;
 
+            if ((*mark == list_last_entry(head, typeof(*le), le_link)) && stats->kst_keys == 0)
+                ++runlen_max;
+
```

Comment:
Seems like we only need to check (keys == 0) ???

Reply:
The other check ensures that cn_tree_prepare_compaction() sets cw_drop_tombs and there's a higher chance that the k/kv-compact drops all these ptomb-only kvsets. But I think you're right.. Even without the first check, there's value in including all the ptomb-only kvsets in the run as we will end up with a fewer kvsets in the output.